framework/include/cppmicroservices/ServiceInterface.h

```diff
@@ -142,7 +142,8 @@ namespace detail
 /// \cond
 template<class T> std::string us_service_interface_iid()
```

Comment:
In this particular case, I think it makes more sense for the return type to be `const std::string&` to avoid a potential copy. If we do this, we should also look at all `us_service_interface_iiid()` calls and use something like
```
auto& clazz = us_service_interface_iid();
```

Reply:
Thanks I have made this particular change.

> you could check if changing std::string to const std::string& is doable without much effort.

Did you mean: store the `string`s in the map in registry as `const std::string &`? Now that you've broached this topic, I am working on an RFC where the idea is to store the keys of the services as numeric IDs instead of the `std::string`s currently. This was suggested by one of our clients and his belief is that the string comparison used when doing map operations is significantly more expensive when compared to IDs. I'll have an RFC up soon.

Reply:
I meant exactly what you did :+1: 
Using integer ids internally sounds a good idea, let's see what you come up with!